[FLINK-21589][docs] Document table pipeline upgrades (#17260)
apache/flink: docs/content/docs/concepts/glossary.md

the data stream. The term "Sub-Task" emphasizes that there are multiple parallel Tasks for the same
[Operator](#operator) or [Operator Chain](#operator-chain).

> #### Table Program

Reviewer: IMHO For each glossary entry there should always be a link to a page with more details about the topic, Perhaps this one https://ci.apache.org/projects/flink/flink-docs-release-1.13/docs/dev/table/common/ for this specific entry?

the data stream. The term "Sub-Task" emphasizes that there are multiple parallel Tasks for the same
[Operator](#operator) or [Operator Chain](#operator-chain).

#### Table Program

> Describes one or more pipelines that are declared with Flink's relational API. It is a generic term
> for both Table API or SQL pipelines which might be used interchangeably.

Reviewer 2: ```suggestion
This is a generic term for pipelines declared with Flink's relational APIs (Table API or SQL).   
```
